msdk-io/msdk-io-csv/src/main/java/io/github/msdk/io/csv/CsvFileExportMethod.java

```diff
@@ -0,0 +1,264 @@
+            return null;
+        }
+
+        // Get number of rows
+        totalRows = featureTable.getRows().size();
+
+        // Write data to CSV file
+        writeData(featureTable, writer);
+
+        // Close file
+        try {
+            writer.close();
+        } catch (Exception e) {
+            logger.info("Could not close file " + csvFile);
+            return null;
+        }
+
+        return csvFile;
+    }
+
+    private void writeData(FeatureTable featureTable, FileWriter writer) {
+
+        // Buffer for writing
+        StringBuffer line = new StringBuffer();
+
+        // Write column headers
+        for (FeatureTableColumn<?> column : columns) {
+            String columnName = column.getName();
+            Sample sample = column.getSample();
+
+            // Add sample name to column name if available
+            if (sample != null) {
+                columnName = sample.getName() + " " + columnName;
+            }
+
+            line.append(escapeStringForCSV(columnName) + separator);
+        }
+
+        // Remove last separator
+        String stringLine = removeLastCharacter(line.toString());
+
+        // Write the line to the CSV file
+        try {
+            writer.write(stringLine);
+        } catch (Exception e) {
+            logger.info("Could not write to file " + csvFile);
+            return;
+        }
+
+        if (canceled)
+            return;
+
+        // Write data values
+        List<FeatureTableRow> rows = featureTable.getRows();
+        for (FeatureTableRow row : rows) {
+
+            // Reset the buffer
+            line.setLength(0);
+
+            // Loop through all columns
+            for (FeatureTableColumn<?> column : columns) {
+                Object object = row.getData(column);
+                String strValue = "";
+
+                if (object == null) {
+                    strValue = "";
+                }
+                // Chromatography Info
+                else if (object instanceof ChromatographyInfo) {
+                    ChromatographyInfo chromatographyInfo = (ChromatographyInfo) object;
+                    Float rt1 = chromatographyInfo.getRetentionTime();
+                    Float rt2 = chromatographyInfo.getSecondaryRetentionTime();
+
+                    if (rt1 != null) {
+                        strValue = rt1.toString();
+                    }
+                    if (strValue != null && rt2 != null) {
+                        strValue = strValue + itemSeparator;
+                    }
+                    if (rt2 != null) {
+                        strValue = strValue + rt2.toString();
+                    }
+                }
+                // Ion annotations
+                else if (object instanceof List<?>) {
+                    List<IonAnnotation> ionAnnotations = (List<IonAnnotation>) object;
+                    for (IonAnnotation ionAnnotation : ionAnnotations) {
```

Comment:
That is not possible, because java generics are recognized only at compilation time, not at runtime. At runtime, all Lists are the same, no matter if they were declared as List<X> or List<Y>.

It is probably safer to iterate over the list and make sure each object you get is actually instanceof IonAnnotation.